### PR TITLE
refactor!: remove module specific config options

### PR DESCRIPTION
**Pull request by TimoGlastra**
Removes the module specific config options from the agent config. They have been deprecated in 0.3.x. The reason for removing them now instead of later is that when a custom module is provided, the config option in the init config doesn't have any effect anymore, which can cause unexpected behvaiour.

### CODE REVIEW

**Review by genaris — APPROVED** [review]
I think it's appropriate to do this step now.

It would be probably nice to continue thinking a bit more about the 'module bundles', Agent Factory or something that could make Agent creation a bit less verbose (at least for the simplest use cases). We are already doing this (sort of) in our tests, so we can probably arrive to something more generalist.

File: packages/anoncreds/src/protocols/proofs/v1/__tests__/v1-connectionless-proofs.e2e.test.ts [hunk 1]
```diff
@@ -388,28 +395,34 @@ describe('V1 Proofs - Connectionless - Indy', () => {
...
-          domain: 'https://example.com',
+        ...getLegacyAnonCredsModules({
+          autoAcceptProofs: AutoAcceptProof.Always,
         }),
-        mediatorPickupStrategy: MediatorPickupStrategy.PickUpV1,
-      },
-      getLegacyAnonCredsModules({
-        autoAcceptProofs: AutoAcceptProof.Always,
-      })
+        mediationRecipient: new RecipientModule({
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/anoncreds/src/protocols/proofs/v1/__tests__/v1-connectionless-proofs.e2e.test.ts:403]: Yes! I can add a separate PR to make this change

[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/anoncreds/src/protocols/proofs/v1/__tests__/v1-connectionless-proofs.e2e.test.ts:403]: Fixed it in this PR

File: packages/core/CHANGELOG.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ See [Conventional Commits](https://conventionalcommits.org) for commit guideline
...
 - **proofs:** proofs module migration script for 0.3.0 ([#1020](https://github.com/hyperledger/aries-framework-javascript/issues/1020)) ([5e9e0fc](https://github.com/hyperledger/aries-framework-javascript/commit/5e9e0fcc7f13b8a27e35761464c8fd970c17d28c))
 - remove keys on mediator when deleting connections ([#1143](https://github.com/hyperledger/aries-framework-javascript/issues/1143)) ([1af57fd](https://github.com/hyperledger/aries-framework-javascript/commit/1af57fde5016300e243eafbbdea5ea26bd8ef313))
-- **routing:** add reconnection parameters to RecipientModuleConfig ([#1070](https://github.com/hyperledger/aries-framework-javascript/issues/1070)) ([d4fd1ae](https://github.com/hyperledger/aries-framework-javascript/commit/d4fd1ae16dc1fd99b043835b97b33f4baece6790))
+- **routing:** add reconnection parameters to MediationRecipientModuleConfig ([#1070](https://github.com/hyperledger/aries-framework-javascript/issues/1070)) ([d4fd1ae](https://github.com/hyperledger/aries-framework-javascript/commit/d4fd1ae16dc1fd99b043835b97b33f4baece6790))
```
Comment (genaris) [inline packages/core/CHANGELOG.md:77]: This should not be changed I think. I guess this change came from a wild search & replace :stuck_out_tongue:

Comment (TimoGlastra) [reply packages/core/CHANGELOG.md:77]: Another 40 minutes of CI :(

**Pull request merged**